src/Patch/Element.js

```diff
@@ -0,0 +1,137 @@
+import Native from './Native';
+import CustomElementInternals from '../CustomElementInternals';
+import * as CESymbols from '../CustomElementInternalSymbols';
+import * as Utilities from '../Utilities';
+
+import PatchParentNode from './Interface/ParentNode';
+import PatchChildNode from './Interface/ChildNode';
+
+/**
+ * @param {!CustomElementInternals} internals
+ */
+export default function(internals) {
+  if (Native.Element_attachShadow) {
+    Utilities.setPropertyUnchecked(Element.prototype, 'attachShadow',
+      /**
+       * @this {Element}
+       * @param {!{mode: string}} init
+       * @return {ShadowRoot}
+       */
+      function(init) {
+        const shadowRoot = Native.Element_attachShadow.call(this, init);
+        this[CESymbols.shadowRoot] = shadowRoot;
+        return shadowRoot;
+      });
+  } else {
+    console.warn('Custom Elements: `Element#attachShadow` was not patched.');
+  }
+
+  // TODO: Patch instances in browsers without an `Element#innerHTML` descriptor (Early Chrome, IE).
+  if (Native.Element_innerHTML) {
+    Object.defineProperty(Element.prototype, 'innerHTML', {
+      enumerable: Native.Element_innerHTML.enumerable,
+      configurable: true,
+      get: Native.Element_innerHTML.get,
+      set: function(htmlString) {
+        Native.Element_innerHTML.set.call(this, htmlString);
+        internals.patchTree(this);
+        internals.upgradeTree(this);
+        return htmlString;
+      },
+    });
+  } else {
+    console.warn('Custom Elements: `Element#innerHTML` was not patched.');
+  }
+
+  Utilities.setPropertyUnchecked(Element.prototype, 'setAttribute',
+    /**
+     * @this {Element}
+     * @param {string} name
+     * @param {string} newValue
+     */
+    function(name, newValue) {
+      const oldValue = Native.Element_getAttribute.call(this, name);
```

Comment:
To improve perf for all these patches, suggest checking first if the element is custom and if not do no other work than calling the native function.